src/main/java/org/nuxeo/binary/metadata/internals/BinaryMetadataServiceImpl.java

```diff
@@ -199,7 +200,11 @@ public void writeMetadata(DocumentModel doc, CoreSession session, String mapping
                         && !(metadataValue.getClass().isArray())) {
                     metadataValue = metadataValue.toString();
                 }
-                doc.setPropertyValue(metadataMapping.get(metadata), (Serializable) metadataValue);
+                try {
+                    doc.setPropertyValue(metadataMapping.get(metadata), (Serializable) metadataValue);
+                } catch (PropertyException e) {
+                    log.error("Failed to set property value for metadata: " + metadata);
```

Comment:
It would be better to use WARN at most, and also log the value of `metadataMapping.get(metadata)` and `metadataValue`. In addition, for people to be able to know what's going on, I would log the doc id and path and maybe title, and also the blob's xpath.